### ISSUE TITLE
Unsized rvalues now making this lint inaccurate

### ISSUE THREAD

**Issue by earthengine** [rust-lang-nursery/rust-clippy#3420]
The following code

```rust
#![feature(unsized_locals)]
fn a(f: Box<dyn FnOnce()>) {
    f()
}

fn main() {
    let f = || println!("Hello!");
    a(Box::new(f))
}
```
[Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2015&gist=947218014e7f41d16d90d5372432e2ce)

is now allowed in nightly. However, Clippy gives inaccurate lint:

```rust
    Checking playground v0.0.1 (/playground)
warning: this argument is passed by value, but not consumed in the function body
 --> src/main.rs:2:9
  |
2 | fn a(f: Box<dyn FnOnce()>) {
  |         ^^^^^^^^^^^^^^^^^ help: consider taking a reference instead: `&Box<dyn FnOnce()>`
  |
  = note: #[warn(clippy::needless_pass_by_value)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#needless_pass_by_value

    Finished dev [unoptimized + debuginfo] target(s) in 0.66s
```

Because when run the code above, it do consumes the variable `f` when using unsized rvalues, and it cannot run without it.

**Comment by nk9** [comment]
Looks like this feature is being/has been[ de-RFCed](https://github.com/rust-lang/rfcs/pull/3829)? (TIL de-RFCing is a thing.) Anyway, the feature has been [removed from nightly](https://github.com/rust-lang/rust/issues/111942). When I use the code above in nightly, clippy doesn't even get to run:

```
error[E0557]: feature has been removed
 --> src/main.rs:1:15
  |
1 |    #![feature(unsized_locals)]
  |               ^^^^^^^^^^^^^^ feature has been removed
  |
  = note: removed in 1.89.0
  = note: removed due to implementation concerns; see https://github.com/rust-lang/rust/issues/111942
```

Seems like this can be closed.

@rustbot label S-triage
